src/mscorlib/shared/System/Globalization/IdnMapping.cs

```diff
@@ -156,6 +157,14 @@ public override int GetHashCode()
             return (_allowUnassigned ? 100 : 200) + (_useStd3AsciiRules ? 1000 : 2000);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe string GetStringForOutput(string originalString, char* input, int inputLength, char* output, int outputLength)
+        {
+            return originalString.Length == inputLength && new ReadOnlySpan<char>(input, inputLength).SequenceEqual(new ReadOnlySpan<char>(output, outputLength)) ?
```

Comment:
@tarekgh, is there any situation where the conversion would yield a different result but where the lengths would still match?  I'm wondering if we could get rid of the SequenceEqual and just make it:
```C#
return originalString.Length == inputLength && inputLength == outputLength ?
    originalString :
    new string(output, 0, outputLength);
```
?

Thanks.

Reply:
I expect the length will be different in most of the cases (if not all) so I think your check of the length would be good enough here.

Reply:
> in most of the cases (if not all)

Thanks, though if it won't definitely be all, we'll want to keep the SequenceEqual.  Can we prove one way or another whether it would be all cases?

Reply:
Usually, when you have non-ASCII character, it will be converted to Punycode which is usually more than one character. so the string length would be different. I don't know a case that can cause producing the same string length.

CC @ShawnSteele if he can advise more about that.

Reply:
@stephentoub don't block on Shawn's reply. I believe your changes are good to go.

Reply:
I believe you can have strings with the same result.  Typically non-ASCII will cause xn-- to be prepended, making it longer, plus some other stuff, however... the normalization step can also throw out characters, which could make the input shorter.  Or combining sequences could normalize to single codepoints, enough of them might make up for the xn-- space.  Certainly 99% of the time theyʻd be different lengths, but...